pkg/k8sutil/k8sutil.go

```diff
@@ -67,13 +67,13 @@ func InitOperatorService() (*v1.Service, error) {
 		Spec: v1.ServiceSpec{
```

Comment:
This comment is for line 53 since I can't comment on that.

The operator should create the metrics service within its own namespace. Right now we're using `GetWatchNamespace()` for that. We've been using that interchangeably for the operator namespace since we set it via the downward API in `deploy/operator.yaml`
```YAML
env:
  - name: WATCH_NAMESPACE
    valueFrom:
      fieldRef:
        fieldPath: metadata.namespace
```
but that might not always be true: users may not set that env, or most likely it will be a set of namespaces in the future #767 

For the operator namespace we can do what the leader election pkg does and read `/var/run/secrets/kubernetes.io/serviceaccount/namespace` to get the namespace (probably reuse that util)
https://github.com/operator-framework/operator-sdk/blob/master/pkg/leader/leader.go#L148
The above util also accounts for when the operator is running locally via `up local` and returns an error so we can skip creating the metrics service(just like we skip leader election).

That way the operator always creates the service in its own namespace, and we can start separating the operator namespace from the watch namespace(s).

Reply:
Makes sense, do you mind if I open an issue and do this in a follow up PR?

Reply:
Yeah we can put a TODO or track this an in issue.
Or it seems like we can rebase this after #836

Reply:
Rebased, so this is done.